src/pages/index.js

```diff
@@ -29,50 +29,165 @@ import {
+      ".cards__list"
+    );
+
+    cardList.renderItems();
+  }
+);
+// =============================================================================
+// Handle Clicks
+// =============================================================================
+
+function handleLikeClick() {
+  if (this._likes) {
+    api
+      .unlikeCard(this._id)
+      .then((res) => {
+        this.setLikes(res.isLiked);
+      })
+      .catch((error) => {
+        console.log(`An error has occured ${error}`);
+      });
+  } else {
+    api
+      .likeCard(this._id)
+      .then((res) => {
+        this.setLikes(res.isLiked);
+      })
+      .catch((err) => console.error(err));
+  }
+}
+
+function handleCardClick(data) {
+  previewImagePopup.open(data);
+}
+
+const deleteCardPopup = new PopupWithConfirmation("#delete-card-modal");
+deleteCardPopup.setEventListeners();
+
+function handleDeleteClick(card) {
+  deleteCardPopup.open();
+  deleteCardPopup.setSubmitAction(() => {
+    deleteCardPopup.setLoading(true);
+    api
+      .deleteCard(card.getId())
+      .then(() => {
+        card.remove();
+        deleteCardPopup.close();
+      })
+      .catch((err) => console.error(err))
+      .finally(() => {
+        deleteCardPopup.setLoading(false, "Yes");
+      });
+  });
+}
+
+function createCard({ name, link, isLiked, _id, userId, ownerId }) {
   const cardElement = new Card(
-    { name, link },
+    { name, link, isLiked, _id, userId, ownerId },
+    userID,
     "#card-template",
-    ({ name, link }) => {
-      previewImagePopup.open({ name, link });
-    }
+    handleCardClick,
+    handleDeleteClick,
+    handleLikeClick
   );
   return cardElement.getView();
 }
-
+// =============================================================================
 // Preview Popup
+// =============================================================================
 
 const previewImagePopup = new PopupWithImage({
   popupSelector: "#preview-modal",
+  handleFormSubmit: handleProfileEditSubmit,
 });
 previewImagePopup.setEventListeners();
 
-//Profile Popup
-
-const newUser = new UserInfo(".profile__title", ".profile__description");
+// =============================================================================
+// Profile Popup
+// =============================================================================
 
-function handleProfileEditSubmit({ title, description }) {
-  newUser.setUserInfo(title, description);
+function handleProfileEditSubmit({ name, about }) {
+  newUser.setUserInfo(name, about);
   profilePopup.close();
 }
 
-const profilePopup = new PopupWithForm(
-  "#profile-edit-modal",
-  handleProfileEditSubmit
-);
+const profilePopup = new PopupWithForm({
+  popupSelector: "#profile-edit-modal",
+  handleFormSubmit: (inputValues) => {
+    profilePopup.showLoading();
+    api
+      .editProfile(inputValues)
+      .then(() => {
+        userInfo.setUserInfo(inputValues);
+        profilePopup.close();
+      })
+      .catch((err) => console.error(err))
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors